homeassistant/components/refoss/switch.py

```diff
@@ -0,0 +1,92 @@
+"""Switch for Refoss."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from refoss_ha.const import DOMAIN, DEVICE_LIST_COORDINATOR, HA_SWITCH, LOGGER
+from refoss_ha.controller.toggle import ToggleXMix
+from refoss_ha.controller.device import BaseDevice
+from .device import MerossDevice
+from . import MerossCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    def entity_adder_callback():
+        coordinator: MerossCoordinator = hass.data[DOMAIN][DEVICE_LIST_COORDINATOR]
+
+        devicelist = coordinator.find_devices()
+
+        new_entities = []
+        try:
+            for device in devicelist:
+                if isinstance(device, ToggleXMix):
+                    if len(device.channels) > 0:
+                        for channel in device.channels:
+                            w = SwitchEntityWrapper(
+                                device=device, channel=channel, coordinator=coordinator
+                            )
+                            if (
+                                w.unique_id
+                                not in hass.data[DOMAIN]["ADDED_ENTITIES_IDS"]
+                            ):
+                                new_entities.append(w)
```

Comment:
```suggestion
            for device in devicelist:
                if not isinstance(device, ToggleXMix):
                    continue
                    
                if len(device.channels) == 0:
                     continue
                     
                for channel in device.channels:
                    w = SwitchEntityWrapper(
                        device=device, channel=channel, coordinator=coordinator
                    )
                    if (
                        w.unique_id
                        not in hass.data[DOMAIN]["ADDED_ENTITIES_IDS"]
                    ):
                        new_entities.append(w)
```

Adding guards instead of nested if statements